src/JsonSchema/SchemaFactory.php

```diff
@@ -235,12 +235,12 @@ private function buildDefinitionName(string $className, string $format = 'json',
         if (null !== $inputOrOutputClass && $className !== $inputOrOutputClass) {
             $parts = explode('\\', $inputOrOutputClass);
             $shortName = end($parts);
-            $prefix .= ':'.$shortName;
+            $prefix .= urlencode(':').$shortName;
         }
 
         if (isset($this->distinctFormats[$format])) {
             // JSON is the default, and so isn't included in the definition name
-            $prefix .= ':'.$format;
+            $prefix .= urlencode(':').$format;
         }
 
         if (isset($serializerContext[DocumentationNormalizer::SWAGGER_DEFINITION_NAME])) {
```

Comment:
I'd encode at the bottom of this function and follow https://github.com/api-platform/core/issues/1969#issuecomment-392033381 by:

- replacing `~` and `/` 
- urlencoding the final `$name`

Reply:
Introduced new method `\ApiPlatform\Core\JsonSchema\SchemaFactory::encodeDefinitionName` which does this.
But need to urlencode first to not overwrite replacing for `~` and `/`.
And it comes out that only the refs should be encoded.